[#4248] Set DamageRoll's options.appearance pre-evaluation

roth-michael opened the pull request:
closes #4248 

Seems straightforward enough. Only potential issues I can think of:
- Will this catch _all_ damage rolls? I'm not intimately familiar with the rolling workflow.
- Would a check be desired for whether either each roll is a `DamageRoll` or whether they have `roll.options.type`? Based on my understanding & testing, if it's undefined the dice will roll with the default appearance.
- Should this be moved down a few lines, so it's inside the `message.create !== false` check? In theory DSN will only roll the dice if they're part of a message, but it felt like I should leverage the existing for loop for this one.

(__Per Zhell's suggestion, moved into `DamageRoll#evaluate` - none of the above are concerns any longer__)

roth-michael committed:
set options.appearance pre-evaluation

krbz999 commented:
Why not inside the `DamageRoll` class?

roth-michael commented:
Checked in there originally, but couldn't find an appropriate place for it. The appearance has to be set after damage type is chosen in the dialog (if any), but before the roll is sent to chat. Since 4.0, I think that all happens in the `BasicRoll` class's `build` function.

krbz999 commented:
> Checked in there originally, but couldn't find an appropriate place for it. The appearance has to be set after damage type is chosen in the dialog (if any), but before the roll is sent to chat. Since 4.0, I think that all happens in the `BasicRoll` class's `build` function.

Without having looked at it, `DamageRoll#evaluate` may be the place.

roth-michael committed:
move appearence set into DamageRoll#evaluate

roth-michael commented:
Yup. Duh. Brain said "I don't see evaluate function so I can't override."

arbron commented on module/dice/damage-roll.mjs:
```diff
@@ -298,4 +298,14 @@ export default class DamageRoll extends BasicRoll {
+
+  /** @inheritDoc */
+  async evaluate(options={}) {
+    this.options.appearance = { colorset: this.options.type };
```
It would probably be good to have this behind a check and to use `setProperty` to avoid deleting any existing appearance options:

```suggestion
    if ( this.options.type && !this.options.appearance?.colorset ) {
      foundry.utils.setProperty(this.options, "appearance.colorset", this.options.type);
    }
```

arbron commented:
@roth-michael Give a comment on the original issue so I can assign it to you

roth-michael committed:
requested change

Co-authored-by: Jeff Hitchcock <github444@command-center.org>

roth-michael commented:
`evaluateSync`, if for whatever reason called, should never result in a DSN dice roll, right? There's no weird situation I’m not thinking of that would necessitate an override of that method too?

krbz999 commented:
> `evaluateSync`, if for whatever reason called, should never result in a DSN dice roll, right? There's no weird situation I’m not thinking of that would necessitate an override of that method too?

Sync rolls cannot involve dice.

roth-michael commented:
That was my understanding as well, just wanted to verify. Thanks.

thatlonelybugbear committed:
[#5683] Deprecated "Token" (#5684)

Fyorl committed:
[#5705] Fix adjusting item & activity uses on actor sheets.

arbron committed:
[#5708] Always populate message data in `BasicRoll#buildPost`

Always completely build out the full chat message data even if
`create` is set to `false` so that it can be used to manually
create the message later.

Closes #5708

arbron committed:
[#5709] Fix double `submit` events in `Dialog5e`

Removes the extra code needed to handle the `<form>` being the
content element rather than the outer element which was causing
the form submission handler from being called twice.

Closes #5709

MaxPat931 committed:
[#5204] Use Token Name instead of Actor ProtoToken Name for Combat Groups (#5502)

Co-authored-by: Jeff Hitchcock <github444@command-center.org>
Co-authored-by: Kim Mantas <kim.mantas@gmail.com>

Fyorl committed:
Tidy up deprecated uses of FormDataExtended.

Fyorl committed:
Clean up TODO.

Fyorl committed:
[#5496] Fix group actor sheet closing when a group member is deleted.

Fyorl committed:
[#4882] Prevent damage type parts from totalling below 0.

Fyorl committed:
[#5377] Fix legendary actions and legendary resistances not recovering on a long rest.

Fyorl committed:
Remove another TODO.

Fyorl committed:
[#5401] Fix nonmagical items still requiring attunement in some cases.

Fyorl committed:
[#5378] Fix subtitle reading 'undefined' for effects rendered in effects tray.

Fyorl committed:
Check magical property is applicable to items before denying attunement.

Fyorl committed:
[#5608] Avoid extra rollData collection. Ensure magical bonus is applied to damage labels.

Fyorl committed:
[#5676] Fix migration notification staying at 0% for Worlds without any Documents.

Fyorl committed:
[#5677] Fix newly-dropped spells not inheriting preparation mode from the section they were dropped on.

arbron committed:
Update system.json for 5.1.0 release

arbron committed:
[#5702] Apply proper roll mode for fast-forwarded rolls

Closes #5702

arbron committed:
[#5698] Fix toggling favorite effects, dropping activity favorites

Closes #5698

arbron committed:
[#5693] Fix temp max HP from making max hit points negative

Closes #5693

arbron committed:
Re-extract compendium content after V13 migration

arbron committed:
[#4553] Clean up elementals in the 2014 SRD

arbron committed:
[#4553] Clean up fey in the 2014 SRD

arbron committed:
[#4553] Clean up fields in the 2014 SRD

Fyorl committed:
[#2803] Prevent drags within input elements from dragging the whole element.

Fyorl committed:
[#4727] [#5692] Set spell source class appropriately during Advancement.

Fyorl committed:
[#5689] Fix spell cast time not including ritual information when displayed in certain places.

Fyorl committed:
[#5679] Fix error thrown when toggling condition for the first time after sheet render.

Fyorl committed:
[#5706] Fix failing to clear prepared status when dropping a spell onto an Actor sheet.

vszalai committed:
[#3386] [#5707] Added options for resetting temp max and temp hp during a rest (#5623)

Co-authored-by: Kim Mantas <kim.mantas@gmail.com>

arbron committed:
[#5714] Fix bastion description not being editable/viewable

Swapped the bastion description to use the `<prose-mirror>`
element since the `{{ editor }}` helper doesn't work with the new
sheets.

Closes #5714

arbron committed:
[#5513, #5629] Fix issues with critical damage bonuses (#5716)

Ensures all critical damage bonuses either from the weapon or from
the powerful critical setting are tagged as such and adds an extra
step to strip them out before the terms are prepared. This should
resolve the issues with critical bonus damage being doubled when
the attack roll includes a crit.

Closes #5513
Closes #5629

arbron committed:
[#5463] Fix currency conversion errors

To avoid errors resulting from floating point errors this modifies
conversion to instead convert to smallest unit, rather than to the
base unit. This means that the values are always full integers
rather than decimals.

Closes #5463

arbron committed:
[#5717] Fix `SummonActivity` when `create` is `false` in config

Closes #5717

arbron committed:
Split various abstract data models into separate files

arbron committed:
[#3776, #4422, #5109] Improve handling of initiative advantage (#5128)

* [#3776, #4422, #5109] Improve handling of initiative advantage

Adjust the initiative score by +5 or -5 relative to the provided
advantage mode. To fully support this adds roll mode to the
initiative config and deprecated the old initiative advantage flag.
If an actor has the deprecated flag set it will appear in the
config app, but as soon as it is unchecked the option will
disappear. The same is true for the special traits tab.

Adds a new helper to `AdvantageModeField` to help setting the mode
from code. If the actor has the initative advantage flag,
remarkable athlete flag, or the surpised status  then the
initiative roll mode will be adjusted accordingly.

The initiative rolling now reflects the roll mode rather than
using the flags directly.

Closes #3776
Closes #4422
Closes #5109

* [#3776] Add incapacitated & invisible to effects that affect initiative

* [#3776] Migrate old initiative advantage flag

* [#3776] Fully remove `initiativeAdv` flag, clear from NPCs

arbron committed:
[#4774] Add properties to classes with Sidekick property (#5281)

* [#4774] Add properties to classes with Sidekick property

Adds properties to classes and give them support for the "Sidekick"
property. Also sets the initial filter when browsing classes on a
PC to exclude sidekicks (but not locked, in case a DM wants to use
the character sheet for their sidekicks).

* [#4474] Fix localization string

arbron committed:
[#5723] Fix invalid variable in `FormulaField#_applyChange*`

An incorrect variable name was copied over when these methods were
moved into `FormulaField` resulting in a bug if upgrade/downgrade
are used with more than 3 values.

Closes #5723

arbron committed:
Add full Legendary Action description when using modern rules

roth-michael committed:
Bring ActiveEffect5e#isSuppressed in line with core expectations (#5632)

Co-authored-by: Kim Mantas <kim.mantas@gmail.com>

roth-michael committed:
Merge branch '5.1.x' into diceSoNiceColorset

roth-michael committed:
Merge branch '5.1.x' into diceSoNiceColorset

JiDW commented:
Hello! Is this still  on the table? I'm prep'ing for the next major version of Dice So Nice and I have some features planned around damage type but I would need to know if this will eventually makes its way into the code or not to plan around it. 

I see that you have DamageRoll.options.type set to the damage type, I'm down to standardize this into a DsN API for other systems to follow, that could be even cleaner tbh

Thanks!

JiDW commented:
It's me again :wave: 
I've been exploring a lot of systems these last hours to see how they handle damage types, I've come to the conclusion that the best move would be to decline this pull request and let DsN pulls from your DamageRoll.type.

If that works from you, you can decline and close the issue, now or when dsn v6 releases

roth-michael commented:
Closing in favor of DsN handling!
